editor/src/components/canvas/controls/position-outline.tsx

```diff
@@ -182,36 +183,62 @@ const collectPinOutlines = (
   return pins
 }
 
-interface PinOutlineProps {
+export interface PinOutlineProps {
   name: string
   isHorizontalLine: boolean
-  startX: number
-  startY: number
-  size: number
+  startX?: number | string
+  endX?: number | string
+  startY?: number | string
+  endY?: number | string
+  size: number | string
   scale: number
 }
 
 const PinOutlineUnscaledSize = 1
 
-const PinOutline = React.memo((props: PinOutlineProps): JSX.Element => {
+export const PinOutline = React.memo((props: PinOutlineProps): JSX.Element => {
   const colorTheme = useColorTheme()
-  const width = props.isHorizontalLine ? props.size : 0
-  const height = props.isHorizontalLine ? 0 : props.size
+  function numberOrStringToSize(value: number | string): string {
+    return typeof value === 'number' ? `${value}px` : value
+  }
+  const width = numberOrStringToSize(props.isHorizontalLine ? props.size : 0)
+  const height = numberOrStringToSize(props.isHorizontalLine ? 0 : props.size)
   const borderTop = props.isHorizontalLine
     ? `${PinOutlineUnscaledSize / props.scale}px dashed ${colorTheme.primary.value}`
     : 'none'
   const borderLeft = props.isHorizontalLine
     ? 'none'
     : `${PinOutlineUnscaledSize / props.scale}px dashed ${colorTheme.primary.value}`
-  const lineLeft = props.startX - PinOutlineUnscaledSize / 2 / props.scale
-  const lineTop = props.startY - PinOutlineUnscaledSize / 2 / props.scale
+
+  const lineLeft =
+    props.startX == null
+      ? undefined
+      : `calc(${numberOrStringToSize(props.startX)} - ${
+          PinOutlineUnscaledSize / 2 / props.scale
+        }px)`
+  const lineTop =
+    props.startY == null
+      ? undefined
+      : `calc(${numberOrStringToSize(props.startY)} - ${
+          PinOutlineUnscaledSize / 2 / props.scale
+        }px)`
+  const lineRight =
+    props.endX == null
+      ? undefined
+      : `calc(${numberOrStringToSize(props.endX)} - ${PinOutlineUnscaledSize / 2 / props.scale}px)`
+  const lineBottom =
+    props.endY == null
+      ? undefined
+      : `calc(${numberOrStringToSize(props.endY)} - ${PinOutlineUnscaledSize / 2 / props.scale}px)`
```

Comment:
I would extract the ternaries here into a function for better readability

Reply:
Refactored in e722c0823.